### PR TITLE
Graph Application cmd spec update

### PR DESCRIPTION
**Pull request by shuagarw**
•Includes Get Application functionality by ObjectId, AppId, IdentifierUri and DisplayNameStartsWith.
•Displays 'DisplayName','IdentifierUris', 'ReplyUris' values when printing application object.
•Added support to fetch the current user 'me' object to identify correct failure for Guest users while creating an application.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@shuagarw__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (shuagarw). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json [hunk 1]
```diff
@@ -1193,33 +1258,80 @@
       ],
       "description": "Request parameters for create a new application"
     },
-    "Application": {
+    "ApplicationFilter": {
       "properties": {
-        "objectId": {
+        "displayNameStartsWith": {
```
Comment (stankovski) [inline arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json]: This should be just "displayName"

File: arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json [hunk 2]
```diff
@@ -1193,33 +1258,80 @@
...
           "type": "string",
-          "description": "Gets or sets object Id"
+          "description": "Application display name starts with"
```
Comment (stankovski) [inline arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json]: Application display name

File: arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json [hunk 3]
```diff
@@ -1193,33 +1258,80 @@
...
         },
-        "objectType": {
+        "appId": {
           "type": "string",
-          "description": "Gets or sets object type"
+          "description": "ApplicationId"
         },
-        "appId": {
+        "identifierUri": {
```
Comment (stankovski) [inline arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json]: This should be:
```
"identifierUris": {
  "type": "array",
  "items": {
     "type": "string"
  },
  "description": "Identifier Uri"
}
```

File: arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json [hunk 4]
```diff
@@ -1193,33 +1258,80 @@
...
-            "type": "string"
+            "$ref": "#/definitions/Application"
           },
-          "description": "Gets or sets application permissions"
-        },
-        "availableToOtherTenants": {
-          "type": "boolean",
-          "description": "Indicates if the application will be available to other tenants"
+          "description": "Application list."
         }
```
Comment (stankovski) [inline arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json]: Add next link property
```js
"odata.nextLink": {
       "type": "string",
       "description": "Gets or sets the URL to get the next set of results."
}
```

File: arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json [hunk 5]
```diff
@@ -55,6 +55,43 @@
...
+        ],
+        "responses": {
+          "200": {
+            "description": "",
+            "schema": {
+              "$ref": "#/definitions/ApplicationListResult"
+            }
+          }
+        },
+        "x-ms-odata": "#/definitions/ApplicationFilter"
```
Comment (stankovski) [inline arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json]: If it's pageable add:
```js
"x-ms-pageable": {
          "nextLinkName": "odata.nextLink",
          "operationName": "Application_ListNext"
        }
```
and then under "x-ms-paths" add:
```js
"/{tenantID}/{nextLink}?Application_ListNext": {
  "get": {
    "tags": [
      "Application"
    ],
    "operationId": "Application_List",
    "description": "Lists applications by filter parameters.",
    "externalDocs": {
      "url": "http://msdn.microsoft.com/en-us/library/azure/hh974476.aspx"
    },
    "parameters": [
      {
        "name": "$filter",
        "in": "query",
        "required": false,
        "type": "string",
        "description": "The filters to apply on the operarion"
      },
      {
        "$ref": "#/parameters/ApiVersionParameter"
      },
      {
        "$ref": "#/parameters/SubscriptionIdParameter"
      },
      {
        "$ref": "#/parameters/tenantIDInPath"
      }
    ],
    "responses": {
      "200": {
        "description": "",
        "schema": {
          "$ref": "#/definitions/ApplicationListResult"
        }
      }
    },
    "x-ms-pageable": {
      "nextLinkName": "odata.nextLink",
      "operationName": "Application_ListNext"
    }
```

File: arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json (hunk 1 above)
Comment (shuagarw) [inline arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json]: Fixed.

File: arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json (hunk 2 above)
Comment (shuagarw) [inline arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json]: Fixed.

File: arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json (hunk 3 above)
Comment (shuagarw) [inline arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json]: This is a filter param. Its a string here.

File: arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json (hunk 4 above)
Comment (shuagarw) [inline arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json]: I dont think its pageable.

File: arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json (hunk 5 above)
Comment (shuagarw) [inline arm-graphrbac/1.42-previewInternal/swagger/graphrbac.json]: Not pageable.

**Pull request merged**